contracts/utils/ItsHubAddressTracker.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import { IItsHubAddressTracker } from '../interfaces/IItsHubAddressTracker.sol';
+
+abstract contract ItsHubAddressTracker is IItsHubAddressTracker {
+    /// @dev The ITS Hub Address Hash
+    bytes32 public immutable itsHubAddressHash;
+
+    /// @dev we need 3 32 byte slots to store the 65 bytes of the hub address.
+    bytes32 private immutable itsHubAddressPrefix;
+    bytes32 private immutable itsHubAddressMiddle;
+    uint8 private immutable itsHubAddressSuffix;
+
+    constructor(string memory hubAddress) {
+        if (bytes(hubAddress).length != 65) revert InvalidHubAddress();
+        bytes32 itsHubAddressPrefix_;
+        bytes32 itsHubAddressMiddle_;
+        uint8 itsHubAddressSuffix_;
+        assembly {
```

Comment:
should we label assembly blocks as [memory-safe](https://docs.soliditylang.org/en/latest/assembly.html#memory-safety)?